cmd/argocd/commands/admin/settings_test.go

```diff
@@ -243,7 +243,11 @@ func tempFile(content string) (string, io.Closer, error) {
 		_ = os.Remove(f.Name())
 		return "", nil, err
 	}
-	defer f.Close()
+	defer func() {
+		if err = f.Close(); err != nil {
+			fmt.Printf("error closing file %q: %v", f.Name(), err)
```

Comment:
Since this is a test, can you panic instead of just logging the error? That'll help us avoid temp files piling up on any machine that's repeatedly running tests.

Reply:
Updated, thx